port/blueprint-permissions/schema.go

```diff
@@ -0,0 +1,232 @@
+			Required: true,
+		},
+		"entities": schema.SingleNestedAttribute{
+			MarkdownDescription: "Entities permissions to read the blueprint",
+			Required:            true,
+			Attributes: map[string]schema.Attribute{
+				"register": schema.SingleNestedAttribute{
+					Required:   true,
+					Attributes: getAssigneeProps("register"),
+				},
+				"unregister": schema.SingleNestedAttribute{
+					Required:   true,
+					Attributes: getAssigneeProps("unregister"),
+				},
+				"update": schema.SingleNestedAttribute{
+					Required:   true,
+					Attributes: getAssigneeProps("update"),
+				},
+				"update_metadata_properties": schema.SingleNestedAttribute{
+					Required: true,
+					Attributes: map[string]schema.Attribute{
+						"icon": schema.SingleNestedAttribute{
+							Required:   true,
+							Attributes: getAssigneeProps("update $icon metadata"),
+						},
+						"title": schema.SingleNestedAttribute{
+							Required:   true,
+							Attributes: getAssigneeProps("update $title metadata"),
+						},
+						"team": schema.SingleNestedAttribute{
+							Required:   true,
+							Attributes: getAssigneeProps("update $team metadata"),
+						},
+						"identifier": schema.SingleNestedAttribute{
+							Required:   true,
+							Attributes: getAssigneeProps("update $identifier metadata"),
+						},
+					},
+				},
+				"update_properties": schema.MapNestedAttribute{
+					Optional: true,
+					NestedObject: schema.NestedAttributeObject{
+						Attributes: getAssigneeProps("update specific property"),
+					},
+					Validators: []validator.Map{
+						mapvalidator.KeysAre(
+							stringvalidator.RegexMatches(
+								regexp.MustCompile(`^[^\$]`),
+								"`update_properties` can not start with `$`, those are reserved to update_metadata_properties",
+							),
+						),
+					},
+				},
+				"update_relations": schema.MapNestedAttribute{
+					Optional: true,
+					NestedObject: schema.NestedAttributeObject{
+						Attributes: getAssigneeProps("update specific relation"),
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *BlueprintPermissionsResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: BlueprintPermissionsResourceMarkdownDescription,
+		Attributes:          BlueprintPermissionsSchema(),
+	}
+}
+
+var BlueprintPermissionsResourceMarkdownDescription = `
+
+# Blueprint Permissions resource
+
+## Example Usage
+
+### Allow access to all members:
+
+` + "```hcl" + `
+resource "port_blueprint_permissions" "microservices_permissions" {
+  blueprint_identifier = "my_blueprint_identifier"
+  entities= {
+	"register" = {
+		"roles": [
+			"Member",
+		],
+		"users": [],
+		"teams": []
+	},
+	"unregister" = {
+		"roles": [
+			"Member",
+		],
+		"users": [],
+		"teams": []
+	},
+	"update" = {
+		"roles": [
+			"Member",
+		],
+		"users": [],
+		"teams": []
+	},
+	"update_metadata_properties" = {
+		"icon" = {
+			"roles": [
+				"Member",
+			],
```

Comment:
see where you can add "admin"  role example to one of the "actions" so they'll see the diversity

Reply:
Done